mountpoint-s3/src/prefetch/parquet_prefetch.rs

```diff
@@ -0,0 +1,182 @@
+use crate::sync::Arc;
+use async_trait::async_trait;
+use bytes::Bytes;
+use futures::pin_mut;
+use futures::StreamExt;
+use mountpoint_s3_client::types::ETag;
+use mountpoint_s3_client::ObjectClient;
+use parquet::errors::ParquetError;
+use parquet::file::footer::decode_footer;
+use parquet::file::metadata::ParquetMetaData;
+use parquet::file::reader::{ChunkReader, Length};
+use std::collections::HashMap;
+use std::io::{Read, Seek, SeekFrom};
+use tracing::debug_span;
+use tracing::trace;
+
+use super::PrefetchReadError;
+
+pub async fn read_parquet_metadata<Client>(
+    client: Arc<Client>,
+    bucket: &str,
+    key: &str,
+    if_match: ETag,
+    total_size: u64,
+) -> Result<Bytes, PrefetchReadError<Client::ClientError>>
+where
+    Client: ObjectClient + Send + Sync + 'static,
```

Comment:
Keep the key and etag consistent, take a ref of the `ETag` (and then you can avoid the clone in `prefetch.rs`) or make bucket and key `String` (and clone those too in `prefetch.rs`).

```suggestion
pub async fn read_parquet_metadata<Client>(
    client: Arc<Client>,
    bucket: &str,
    key: &str,
    if_match: &ETag,
    total_size: u64,
) -> Result<Bytes, PrefetchReadError<Client::ClientError>>
where
    Client: ObjectClient + Send + Sync + 'static,
```

I don't have a strong preference on which one 😄

Reply:
For sure, I agree with that approach